<a href="https://colab.research.google.com/github/Hyeytt/basic_python221028/blob/main/basic_python221028.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 1. 사용자로부터 두 개의 정수를 받아서 정수의 합, 정수의 차, 정수의 곱, 정수의 펑균, 큰 수 작은 수를 계산하여 화면에 출력하는 프로그랩을 작성하라. 파이썬이 제공하는 내장 함수 max(x, y), min(x, y)을 사용해보자.

In [ ]:
x = int(input("x의 값은? : "))          
y = int(input("y의 값은? : "))  
print("두수의 합은 : ", x+y )         
print("두수의 차는 : ", x-y )
print("두수의 곱은 : ", x*y)
print("두수의 평균은 :" ,(x+y)/2)
print("큰 수는? : " , max(x, y))      
print("작은 수는?: " , min(x,y))      

x의 값은? : 10
y의 값은? : 5
두수의 합은 :  15
두수의 차는 :  5
두수의 곱은 :  50
두수의 평균은 : 7.5
큰 수는? :  10
작은 수는?:  5


# 2. 원기둥의 부피를 계산하는 프로그랩을 작성해보자. 원기둥의 부피는 다음과 같이 계산한다

In [ ]:
r = float(input("r의 값은? : "))
h = float(input("h의 값은? : "))

vol = 3.141592 * r**2 * h
print("원기둥의 부피는" , vol )

r의 값은? : 10
h의 값은? : 10
원기둥의 부피는 3141.592


# 3. 사용자로부터 정수를 받아서 정수의 자리수의 합을 계산하는 프로그램을  작성하여 보자. 예릍 들어서 사용자가 1234를 입력하였다면 1+2+3+4를 계산하면 된다. 나머지 연산자와 정수 나눗셈 연산자 〃를 적극적으로 사용해보자.

In [ ]:
number  = int(input("정수를 입력하시오 : "))

sum = 0     # 초기화 시키기 , 처음에 시작할때 
sum = sum + number % 10
number = number // 10 

sum = sum + number % 10
number = number // 10

sum = sum + number % 10 
number = number // 10 

sum = sum + number % 10
number = number // 10 

print(" 자리수의 합은 : ", str(sum))

정수를 입력하시오 : 1234
 자리수의 합은 :  10


# 4. 사용자로부터 두 점의 좌표 (xl, yl)과 (x2, y2)를 입력받아서 두 점 사이의 거리를 계산하는 프로그램을 작성해보자. 스크립트 모드로 작성하라. 거리는 다음 식으로 계산한다.

In [ ]:
x1 = int(input("x1의 값을 입력하시오 : "))
y1 = int(input("y1의 값을 입력하시오 : "))
x2 = int(input("x2의 값을 입력하시오 : "))
y2 = int(input("y2의 값을 입력하시오 : "))
dist = ((x2 - x1)**2 +(y2 - y1)**2)**0.5
print("두 점 사이의 거리는", dist , "입니다")


x1의 값을 입력하시오 : 0
y1의 값을 입력하시오 : 0
x2의 값을 입력하시오 : 100
y2의 값을 입력하시오 : 100
두 점 사이의 거리는 141.4213562373095 입니다


# 5. 1번 문제에서 계산한 거리가맞는지, 터틀그래픽으로 확인해보자. 거북이를 읜쪽으로 45도회전하여 141만큼 전진시킨다. 다시 거북이를 (0, 0)으로 이동하고 0도를 가리키게 한 후에 100만큼 전진하고 왼쪽으로 90도 회전하여100만큼 전진한다. 화면에 그러진 직선이 일치하는가?

import turtle

 t = turtle.Turtle()


 t.shape("turtle")
 
t.left(45)

 t.fd(141)

 t.setheading(0)

 t.goto(0,0)

 t.fd(100)

 t.lt(90)

 t.fd(100)


    import turtle

    t = turtle.Turtle()

    t.shape("turtle")

    t.left(45)

    t.fd(141)

    t.setheading(0)

    t.goto(0,0)

    t.fd(100)

    t.lt(90)

    t.fd(100)

    t. _screen.exitonclick()

import turtle

t = turtle.Turtle()


t.shape("turtle")

x1 = int(input("x1 : "))


y1 = int(input("y1 : "))


x2 = int(input("x2 : "))


y2 = int(input("y2 : "))



dist =((x1 - x2)** 2  + (y1 - y2)**2 )**0.5


t.up()


t.goto(x1,y1)


t.down

t.goto(x2,y2)

t.write("점의 길이 = ", str(dist))


# 6. 사용자로부터 두 점을 입력받아서 터틀그래픽을 이용하여 두 점을 연결하는 직선을 그린다. 직선의 끝점에 직선의 길이를 계산하여 출력해보자


import turtle

t = turtle.Turtle()


t.shape("turtle")

x1 = int(input("x1 : "))


y1 = int(input("y1 : "))


x2 = int(input("x2 : "))


y2 = int(input("y2 : "))



dist =((x2 - x1)**2 + (y2 - y1)**2)**0.5



t.up()


t.goto(x1,y1)


t.down

t.goto(x2,y2)

t.write("점의 길이 = " + str(dist))
